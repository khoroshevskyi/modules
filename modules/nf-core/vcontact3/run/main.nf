process VCONTACT3_RUN {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/vcontact3:3.1.3--pyhdfd78af_0' :
        'biocontainers/vcontact3:3.1.3--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(fasta)
    tuple val(meta2), path(database)
    val is_nucleotide

    output:
    tuple val(meta), path("${prefix}/"), emit: out
    path "versions.yml"                , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    def input_command = is_nucleotide ? "-n ${fasta}" : "-p ${fasta}"
    """
    vcontact3 \\
        run \\
        $args \\
        -t $task.cpus \\
        ${input_command} \\
        -d ${database} \\
        -o ${prefix}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        vcontact3: \$(vcontact3 version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo $args

    mkdir -p ${prefix}
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        vcontact3: \$(vcontact3 version)
    END_VERSIONS
    """
}
