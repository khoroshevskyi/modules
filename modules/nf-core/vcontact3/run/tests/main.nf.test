
// nf-core modules test vcontact3/run
nextflow_process {

    name "Test Process VCONTACT3_RUN"
    script "../main.nf"
    config './nextflow.config'
    process "VCONTACT3_RUN"

    tag "modules"
    tag "modules_nfcore"
    tag "vcontact3"
    tag "vcontact3/run"

    test("sarscov2 - contigs - nucleotide") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fasta/contigs.fasta', checkIfExists: true),
                ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta2 map
                    file('/workspace/modules/v230-eukar', checkIfExists: true),
                ]
                input[2] = true
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - contigs - protein") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fasta/contigs.fasta', checkIfExists: true),
                ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta2 map
                    file('/workspace/modules/v230-eukar', checkIfExists: true),
                ]
                input[2] = true
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - bam - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fasta/contigs.fasta', checkIfExists: true),
                ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta2 map
                    file('/workspace/modules/test_database_eukaryotes', checkIfExists: true),
                ]
                input[2] = true
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
