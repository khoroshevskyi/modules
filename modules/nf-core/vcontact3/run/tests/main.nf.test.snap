{
    "sarscov2 - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,0b68c9c08e95078e941c0106718a535f"
                ],
                "out": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0b68c9c08e95078e941c0106718a535f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-24T12:51:16.652963733"
    },
    "sarscov2 - contigs - nucleotide": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "HMMprofile.0.3.merged.faa:md5,3383a91613b667590de87cadfa34e993",
                            "HMMprofile.0.3_SqRoot_dist_and_shared_genes.h5:md5,832300d964665d8bf3e0ad9e16a71656",
                            "HMMprofile.0.4.merged.faa:md5,3383a91613b667590de87cadfa34e993",
                            "HMMprofile.0.4_SqRoot_dist_and_shared_genes.h5:md5,8f216cd726f800ef3feda48f220f33f1",
                            "HMMprofile.0.5.merged.faa:md5,3383a91613b667590de87cadfa34e993",
                            "HMMprofile.0.5_SqRoot_dist_and_shared_genes.h5:md5,9cad60ba9023f54415247fc3f24697ba",
                            "HMMprofile.0.6.merged.faa:md5,3383a91613b667590de87cadfa34e993",
                            "HMMprofile.0.6_SqRoot_dist_and_shared_genes.h5:md5,282016e363bf60e203e675c10455f946",
                            "HMMprofile.0.7.merged.faa:md5,3383a91613b667590de87cadfa34e993",
                            "HMMprofile.0.7_SqRoot_dist_and_shared_genes.h5:md5,4ce32936a8d6323a1775dc4808674e39",
                            "HMMprofile_components.h5:md5,4b6296e9e78036905aa1aff199ee0614",
                            "HMMprofile_crosstabs.h5:md5,99d05e002d571dcbfce16fcc20c350f1",
                            "HMMprofile_gene_support.h5:md5,c6e61090538f32c9ee4f67e17f214232",
                            "HMMprofile_mLayer_dist_ntw.h5:md5,3296539d346a72d8ff77300ce80cb6b2",
                            "HMMprofile_mLayer_shared_ntw.h5:md5,eca90352c9803cb1cf6bdd7eb62658b9",
                            "contigs.eukaryotes.gbga.pkl.gz:md5,87d997854348677e7c5cc0507a87b6f1",
                            "contigs.eukaryotes.performance.pkl.gz:md5,a7c47453e29003dde5ba74bde721b6ad",
                            "contigs.eukaryotes.profile.pkl.gz:md5,94fe9e5114ca69cfd52fd546089ad65c",
                            "contigs.eukaryotes.resolver.pkl.gz:md5,144f726f4d0ffd26bd78089dad1a58ef",
                            "contigs.faa:md5,6b7b89e4a82cceb19ec2d3dcde53797c",
                            "contigs.g2g.parquet:md5,d8449755413d98b7db195bf0d65d7ebb",
                            "contigs.nucl.parquet:md5,5a3a4de3498a7d3e6e4bdb98468de998",
                            [
                                "final_assignments.csv:md5,1e358dba08adce567ccc8e467af6b34f",
                                "performance_metrics.csv:md5,5595902d63bf9ae231a9288760493a68"
                            ]
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,0b68c9c08e95078e941c0106718a535f"
                ],
                "out": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "HMMprofile.0.3.merged.faa:md5,3383a91613b667590de87cadfa34e993",
                            "HMMprofile.0.3_SqRoot_dist_and_shared_genes.h5:md5,832300d964665d8bf3e0ad9e16a71656",
                            "HMMprofile.0.4.merged.faa:md5,3383a91613b667590de87cadfa34e993",
                            "HMMprofile.0.4_SqRoot_dist_and_shared_genes.h5:md5,8f216cd726f800ef3feda48f220f33f1",
                            "HMMprofile.0.5.merged.faa:md5,3383a91613b667590de87cadfa34e993",
                            "HMMprofile.0.5_SqRoot_dist_and_shared_genes.h5:md5,9cad60ba9023f54415247fc3f24697ba",
                            "HMMprofile.0.6.merged.faa:md5,3383a91613b667590de87cadfa34e993",
                            "HMMprofile.0.6_SqRoot_dist_and_shared_genes.h5:md5,282016e363bf60e203e675c10455f946",
                            "HMMprofile.0.7.merged.faa:md5,3383a91613b667590de87cadfa34e993",
                            "HMMprofile.0.7_SqRoot_dist_and_shared_genes.h5:md5,4ce32936a8d6323a1775dc4808674e39",
                            "HMMprofile_components.h5:md5,4b6296e9e78036905aa1aff199ee0614",
                            "HMMprofile_crosstabs.h5:md5,99d05e002d571dcbfce16fcc20c350f1",
                            "HMMprofile_gene_support.h5:md5,c6e61090538f32c9ee4f67e17f214232",
                            "HMMprofile_mLayer_dist_ntw.h5:md5,3296539d346a72d8ff77300ce80cb6b2",
                            "HMMprofile_mLayer_shared_ntw.h5:md5,eca90352c9803cb1cf6bdd7eb62658b9",
                            "contigs.eukaryotes.gbga.pkl.gz:md5,87d997854348677e7c5cc0507a87b6f1",
                            "contigs.eukaryotes.performance.pkl.gz:md5,a7c47453e29003dde5ba74bde721b6ad",
                            "contigs.eukaryotes.profile.pkl.gz:md5,94fe9e5114ca69cfd52fd546089ad65c",
                            "contigs.eukaryotes.resolver.pkl.gz:md5,144f726f4d0ffd26bd78089dad1a58ef",
                            "contigs.faa:md5,6b7b89e4a82cceb19ec2d3dcde53797c",
                            "contigs.g2g.parquet:md5,d8449755413d98b7db195bf0d65d7ebb",
                            "contigs.nucl.parquet:md5,5a3a4de3498a7d3e6e4bdb98468de998",
                            [
                                "final_assignments.csv:md5,1e358dba08adce567ccc8e467af6b34f",
                                "performance_metrics.csv:md5,5595902d63bf9ae231a9288760493a68"
                            ]
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0b68c9c08e95078e941c0106718a535f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-24T13:41:44.590413"
    },
    "sarscov2 - contigs - protein": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "HMMprofile.0.3.merged.faa:md5,1271a1b7a88774c0de6c30c11a856d0f",
                            "HMMprofile.0.3_SqRoot_dist_and_shared_genes.h5:md5,832300d964665d8bf3e0ad9e16a71656",
                            "HMMprofile.0.4.merged.faa:md5,1271a1b7a88774c0de6c30c11a856d0f",
                            "HMMprofile.0.4_SqRoot_dist_and_shared_genes.h5:md5,8f216cd726f800ef3feda48f220f33f1",
                            "HMMprofile.0.5.merged.faa:md5,1271a1b7a88774c0de6c30c11a856d0f",
                            "HMMprofile.0.5_SqRoot_dist_and_shared_genes.h5:md5,9cad60ba9023f54415247fc3f24697ba",
                            "HMMprofile.0.6.merged.faa:md5,1271a1b7a88774c0de6c30c11a856d0f",
                            "HMMprofile.0.6_SqRoot_dist_and_shared_genes.h5:md5,282016e363bf60e203e675c10455f946",
                            "HMMprofile.0.7.merged.faa:md5,1271a1b7a88774c0de6c30c11a856d0f",
                            "HMMprofile.0.7_SqRoot_dist_and_shared_genes.h5:md5,4ce32936a8d6323a1775dc4808674e39",
                            "HMMprofile_components.h5:md5,4d61fdedf6fbb705526092490922fc7d",
                            "HMMprofile_crosstabs.h5:md5,99d05e002d571dcbfce16fcc20c350f1",
                            "HMMprofile_gene_support.h5:md5,3fa4a3add16a520f2fa1329b63c5521a",
                            "HMMprofile_mLayer_dist_ntw.h5:md5,3296539d346a72d8ff77300ce80cb6b2",
                            "HMMprofile_mLayer_shared_ntw.h5:md5,eca90352c9803cb1cf6bdd7eb62658b9",
                            "contigs.eukaryotes.gbga.pkl.gz:md5,8b4560dbeb3b4fa3cb1b221cf9884845",
                            "contigs.eukaryotes.performance.pkl.gz:md5,a7c47453e29003dde5ba74bde721b6ad",
                            "contigs.eukaryotes.profile.pkl.gz:md5,ad9ef2f2ba70badcd5e7b9f2af276e8b",
                            "contigs.eukaryotes.resolver.pkl.gz:md5,144f726f4d0ffd26bd78089dad1a58ef",
                            "contigs.faa:md5,092c56f6c39675749efa0f1684882fc3",
                            "contigs.g2g.parquet:md5,ec9bd18c3a793b76d119d75e645f053f",
                            "contigs.nucl.parquet:md5,df47d594e1038319c21bb379030354e8",
                            [
                                "final_assignments.csv:md5,1e358dba08adce567ccc8e467af6b34f",
                                "performance_metrics.csv:md5,5595902d63bf9ae231a9288760493a68"
                            ]
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,0b68c9c08e95078e941c0106718a535f"
                ],
                "out": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "HMMprofile.0.3.merged.faa:md5,1271a1b7a88774c0de6c30c11a856d0f",
                            "HMMprofile.0.3_SqRoot_dist_and_shared_genes.h5:md5,832300d964665d8bf3e0ad9e16a71656",
                            "HMMprofile.0.4.merged.faa:md5,1271a1b7a88774c0de6c30c11a856d0f",
                            "HMMprofile.0.4_SqRoot_dist_and_shared_genes.h5:md5,8f216cd726f800ef3feda48f220f33f1",
                            "HMMprofile.0.5.merged.faa:md5,1271a1b7a88774c0de6c30c11a856d0f",
                            "HMMprofile.0.5_SqRoot_dist_and_shared_genes.h5:md5,9cad60ba9023f54415247fc3f24697ba",
                            "HMMprofile.0.6.merged.faa:md5,1271a1b7a88774c0de6c30c11a856d0f",
                            "HMMprofile.0.6_SqRoot_dist_and_shared_genes.h5:md5,282016e363bf60e203e675c10455f946",
                            "HMMprofile.0.7.merged.faa:md5,1271a1b7a88774c0de6c30c11a856d0f",
                            "HMMprofile.0.7_SqRoot_dist_and_shared_genes.h5:md5,4ce32936a8d6323a1775dc4808674e39",
                            "HMMprofile_components.h5:md5,4d61fdedf6fbb705526092490922fc7d",
                            "HMMprofile_crosstabs.h5:md5,99d05e002d571dcbfce16fcc20c350f1",
                            "HMMprofile_gene_support.h5:md5,3fa4a3add16a520f2fa1329b63c5521a",
                            "HMMprofile_mLayer_dist_ntw.h5:md5,3296539d346a72d8ff77300ce80cb6b2",
                            "HMMprofile_mLayer_shared_ntw.h5:md5,eca90352c9803cb1cf6bdd7eb62658b9",
                            "contigs.eukaryotes.gbga.pkl.gz:md5,8b4560dbeb3b4fa3cb1b221cf9884845",
                            "contigs.eukaryotes.performance.pkl.gz:md5,a7c47453e29003dde5ba74bde721b6ad",
                            "contigs.eukaryotes.profile.pkl.gz:md5,ad9ef2f2ba70badcd5e7b9f2af276e8b",
                            "contigs.eukaryotes.resolver.pkl.gz:md5,144f726f4d0ffd26bd78089dad1a58ef",
                            "contigs.faa:md5,092c56f6c39675749efa0f1684882fc3",
                            "contigs.g2g.parquet:md5,ec9bd18c3a793b76d119d75e645f053f",
                            "contigs.nucl.parquet:md5,df47d594e1038319c21bb379030354e8",
                            [
                                "final_assignments.csv:md5,1e358dba08adce567ccc8e467af6b34f",
                                "performance_metrics.csv:md5,5595902d63bf9ae231a9288760493a68"
                            ]
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0b68c9c08e95078e941c0106718a535f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-24T13:48:47.964447551"
    }
}